inspirehep/dojson/hep/fields/bd90x99x.py

```diff
@@ -123,22 +93,40 @@ def get_value(value):
 @utils.filter_values
 def references2marc(self, key, value):
     """Produce list of references."""
+    repnos = value.get('arxiv_eprints', [])
+    if 'reportnumber' in value.get('publication_info', {}):
+        repnos.append(value['publication_info']['reportnumber'])
+    j_title = value.get('publication_info', {}).get('journal_title')
+    j_vol = value.get('publication_info', {}).get('journal_volume')
+    j_pg_s = value.get('publication_info', {}).get('page_start')
+    j_pg_e = value.get('publication_info', {}).get('page_end')
+    j_artid = value.get('publication_info', {}).get('artid')
+    pubnote = ''
+    if j_title and j_vol:
+        pubnote = '{},{}'.format(j_title, j_vol)
+        if j_pg_s and j_pg_e:
+            pubnote += ',{}-{}'.format(j_pg_s, j_pg_e)
+        elif j_pg_s:
+            pubnote += ',{}'.format(j_pg_s)
+        if j_artid and j_artid != j_pg_s:
+            pubnote += ',{}'.format(j_artid)
```

Comment:
Can you check if this logic is repeated somewhere else? This looks generally useful.